pingpong/msg_pingpong.c

```diff
@@ -278,8 +278,15 @@ static int run(void)
 	}
 
 	ret = wait_for_completion(txcq, max_credits - credits);
+	if (ret)
+		goto out;
+
 	/* Finalize before closing ep */
-	ft_finalize(fi, ep, txcq, rxcq, FI_ADDR_UNSPEC);
+	if (opts.dst_addr)
+		ret = ft_client_finalize(fi, ep, txcq, FI_ADDR_UNSPEC);
+	else
+		ret = ft_server_finalize(rxcq);
+
```

Comment:
You made this same change 10 different times. Any difference in behavior between the client and server sides should be handled within ft_finalize() itself.